src/components/ui/coaching-sessions/coaching-session-title.tsx

```diff
@@ -49,10 +55,40 @@ const CoachingSessionTitle: React.FC<{
 
   const displayTitle = sessionTitle?.title || defaultSessionTitle().title;
 
+  // Helper to get presence by role
+  const getPresenceByRole = (role: 'coach' | 'coachee'): UserPresence | undefined => {
+    if (!presenceState) return undefined;
+    return Array.from(presenceState.users.values()).find(u => u.role === role);
+  };
```

Comment:
Two considerations here:
1. `role` should be an enum type with two valid options instead of defining it inline here as an ad-hoc type. Perhaps you can think about consolidating this into a type that lives under another appropriate model, or at least gets consolidated into a local type that's calculated at runtime but in one central place under `src/types/`

2.   Performance considerations:
```typescript
return Array.from(presenceState.users.values()).find(u => u.role === role);
```
  - Converting Map to Array on every render could be expensive with many users
  - Consider memoization or direct Map access pattern